django/thunderstore/modpacks/tasks.py

```diff
@@ -0,0 +1,50 @@
+import base64
+import io
+import zipfile
+import yaml
+
+from celery import shared_task
+from typing import Dict
+from thunderstore.core.settings import CeleryQueues
+from thunderstore.modpacks.models.legacyprofile import LegacyProfile, LegacyProfileMetaData
+
+def parse_package_name_and_team(s: str):
+    reversed_splitted = "".join(reversed(s)).split("-", maxsplit=1)
+    name = "".join(reversed(reversed_splitted[0]))
+    team = "".join(reversed(reversed_splitted[1]))
+    return (name, team)
+
+
+def generate_mod_metadata(mod: Dict):
```

Comment:
I'd suggest returning a strongly typed pydantic model from this function.

Reply:
Also I'd suggest having the input be parsed as a pydantic model instead of just an arbitrary dict that'll crash with key access error if something doesn't exist. Let's you make the rest of the code type safe and stop dealing with index accessors as you have the variables as actual fields then.